### PR TITLE
user auth created but requires refactoring, full testing and a new cr…

### PR DESCRIPTION
**Pull request by LSB616**
…eate tech route

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -62,164 +62,164 @@ describe("GET /api/users", () => {
...
+//     return request(app)
+//       .post("/api/technicians")
+//       .send(newTechnician)
+//       .expect(400)
+//       .then(({ body }) => {
+//         const { msg } = body;
+//         expect(msg).toBe("Bad request");
+//       });
+//   });
+// });
```
Comment (stormytuna) [inline __tests__/app.test.js:222]: Why is this commented out?

File: .env [hunk 2]
```diff
@@ -0,0 +1 @@
+JWT_SECRET = abc123
```
Comment (stormytuna) [inline .env:1]: Is this meant to be public?

File: app.js [hunk 3]
```diff
@@ -53,13 +57,18 @@ app.get("/api/users/:user_id/reviews", getUserReviews);
...
 app.patch("/api/users/:user_id/reviews/:review_id", patchUserReview);
 app.patch("/api/users/:user_id/orders/:order_id", patchOrder);
-app.post("/api/users", postUser);
 app.post("/api/users/:user_id/reviews", postReview);
 app.post("/api/users/:user_id/orders", postOrder);
 app.delete("/api/users/:user_id", deleteUser);
 app.delete("/api/:user_id/reviews/:review_id", removeReview);
 app.delete("/api/users/:user_id/orders/:order_id", removeOrder);
 
+app.post("/api/users/register", registerUser);
```
Comment (stormytuna) [inline app.js:66]: Why are we changing this endpoint name?

File: controllers/authMiddleware.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+        } catch (error) {
+            console.log(error)
+            res.status(401)
+            throw new Error('Not authorized')
+        }
+
+        if(!token) {
+            res.status(401)
+            throw new Error('Not authorized, no token')
```
Comment (stormytuna) [inline controllers/authMiddleware.js:25]: Shouldn't these follow our bad request template?

File: db/data/users.js [hunk 5]
```diff
@@ -29,13 +29,18 @@ const userSchema = new mongoose.Schema({
     },
     email: {
       type: String,
+      required: [true, 'Please add a email'],
```
Comment (stormytuna) [inline db/data/users.js:32]: We already have a validation message for this

**Review by stormytuna — COMMENTED** [review]
I trust that it works, just a few things id like clarification on

**Comment by stormytuna** [comment]
Also just realised you'll need to update `api.json`

File: .env (hunk 2 above)
Comment (LSB616) [reply .env:1]: yeah, it doesn't matter too much what that's set to

File: app.js (hunk 3 above)
Comment (LSB616) [reply app.js:66]: was basically to differntiate between api/users/reg and api/tech/reg if we going down that route. It also reflects better on what the end point is doing.

File: controllers/authMiddleware.js (hunk 4 above)
Comment (LSB616) [reply controllers/authMiddleware.js:25]: They can do that also but they are set up to give errors on the front end when I'm hooking it up.

File: db/data/users.js (hunk 5 above)
Comment (LSB616) [reply db/data/users.js:32]: I didn't see that but it doesn't really hurt to leave it in.

**Pull request merged**